docx/text/paragraph.py

```diff
@@ -453,70 +467,76 @@ def _inner_get_tabstops(obj):
             _inner_get_tabstops(para)
             return tabstops
 
+        def apply_formatting(source, first_line_indent=None, left_indent=None):
+            if source:
+                if getattr(source, 'first_line_indent', None) is not None:
+                    first_line_indent = getattr(source, 'first_line_indent')
+                if getattr(source, 'left_indent', None) is not None:
+                    left_indent = getattr(source, 'left_indent')
+            return first_line_indent, left_indent
+
+        # apply paragraph styles by priority (from lowest to highest)
+        first_line_indent = get_base_style_attr(self.style, 'first_line_indent')
+        left_indent = get_base_style_attr(self.style, 'left_indent')
+        first_line_indent, left_indent = apply_formatting(self.style_numbering_format, first_line_indent, left_indent)
+        first_line_indent, left_indent = apply_formatting(self.style.paragraph_format, first_line_indent, left_indent)
+        first_line_indent, left_indent = apply_formatting(self.para_numbering_format, first_line_indent, left_indent)
+        first_line_indent, left_indent = apply_formatting(self.paragraph_format, first_line_indent, left_indent)
```

Comment:
since this part is extremely important for understanding how indentation works, would it make sense to add a couple of comments between lines briefly explaining what this part does?